language-support/ts/codegen/tests/ts/build-and-lint-test/src/__tests__/test.ts

```diff
@@ -348,15 +348,15 @@ test("interfaces", async () => {
   }
   const ifaceContract = await aliceLedger.create(buildAndLint.Main.Asset, assetPayload);
   expect(ifaceContract.payload).toEqual(assetPayload);
-  const [ifaceContract1, events1] = await aliceLedger.exercise(buildAndLint.Main.Asset.Transfer, ifaceContract.contractId, {newOwner: BOB_PARTY});
+  const [ifaceContract1, events1] = await aliceLedger.exercise(buildAndLint.Main.Token.Transfer, ifaceContract.contractId, {newOwner: BOB_PARTY});
   expect(events1).toMatchObject(
```

Comment:
we should also extend this test to exercise the choice both via the interface and via the template. Not quite sure if that passes the JSON api. probably not so assert on the failure.

Reply:
One could try to cast the contract id to template/interface contract id. but the fixed choice is now defined in the interface and not part of the template anymore. We could try to add it back to the template, but that means we would need to traverse all implementations of a template and then generate those choices as part of the template. I'd prefer the choice only defined as part of the interface.

Reply:
Going back to the “archive as an interface” example, that behavior seems somewhat confusing to me: I wouldn’t expect to have to cast to an interface to call that and we deliberately decided that the ledger API should not require this. So having the typescript bindings support this seems like the only consistent option, otherwise we should revisit the choice to allow it on the ledger API.

Reply:
I managed to find a way so that both, the interface object, as well as the template object, have the interface fixed choices available. So this now works both ways.